src/ty/mod.rs

```diff
@@ -103,6 +103,26 @@ impl Type {
 	}
 }
 
+impl<T> Type<T>
+where
+	T: Form,
+{
+	/// Returns the path of the type
+	pub fn path(&self) -> &Path<T> {
+		&self.path
+	}
+
+	/// Returns the generic type parameters of the type
+	pub fn type_params(&self) -> &[T::Type] {
+		&self.type_params
+	}
+
+	/// Returns the definition of the type
+	pub fn type_def(&self) -> &TypeDef<T> {
+		&self.type_def
+	}
```

Comment:
Why getters for `Type` whereas you have `pub` fields everywhere else?

Reply:
Not that I dislike this. :sweat_smile:

Reply:
Actually `pub` fields are gone, replaced by getters